test/dbSpec.js

```diff
@@ -329,6 +329,64 @@ describe('Connection', () => {
             expect(error).toEqual(new Error($text.poolDestroyed));
         });
     });
+
+    describe('db side closing of the connection pool', () => {
+        const singleCN = JSON.parse(JSON.stringify(dbHeader.cn)); // dumb connection cloning;
+        singleCN.max = 1;
+        const dbSingleCN = pgp(singleCN);
+
+        let error;
+
+        beforeEach(done => {
+            dbSingleCN.connect()
+                .then(obj => {
+
+                    const q = obj.any('SELECT pg_sleep(1);');
+
+                    // Terminate all connections after a short delay
+                    promise.delay(100).then(() => {
```

Comment:
If the `delay(100)` is removed this still fails, but that seems to be a different bug :-(